app/src/org/commcare/activities/InstallArchiveActivity.java

```diff
@@ -80,7 +82,7 @@ protected void onCreate(Bundle savedInstanceState) {
         });
 
         btnInstallArchive.setOnClickListener(v -> {
-            String archivePath = editFileLocation.getText().toString();
+            String archivePath = selectedFileUri.toString();
             HiddenPreferences.setLastKnownCczLocation(archivePath);

```

Comment:
We are [operating](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/recovery/measures/ScanCczTask.java#L94) on the last known ccz location as new File(`uri`) , How does that work ?

Reply:
Ohh I missed that.